runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/DokkaProperty.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.dokka.gradle
+
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.kotlin.dsl.property
+import org.jetbrains.dokka.utilities.cast
+import kotlin.reflect.typeOf
+
+internal inline fun <reified T : Any> ObjectFactory.safeProperty() = property<T?>()
```

Comment:
Why do you need these extension methods and can not use `property<T>()` directly? It's already `<T>` not `<T?>`.

Reply:
This is an interesting question: 
So the issue that I have with properties is, that it was really hard for me to track whether or not I expect some value being present in a Property, or if it is actually, truly "optional". The current implementation of `Property` also does not care. It is like "can be there, might not". 

Therefore, I started tracking my expectations of any `Property` myself in the Type-System: 
The convention internally here is: 

- `Property<T?>` -> No value required
- `Property<T>`  -> Fail if value is missing

See this  `getSafe` method, that I am using everywhere. This will use `orNull` if the type T is optional and `get` otherwise.

```
@OptIn(ExperimentalStdlibApi::class)
internal inline fun <reified T> Provider<T>.getSafe(): T =
    if (typeOf<T>().isMarkedNullable) orNull as T
    else get()

```